lib/widgets/message_list.dart

```diff
@@ -194,6 +233,36 @@ class _MessageListState extends State<MessageList> {
   }
 }
 
+class ScrollToBottomButton extends StatelessWidget {
+  const ScrollToBottomButton({super.key, required this.scrollController, required this.visibleValue});
+  final ValueNotifier<bool> visibleValue;
+  final ScrollController scrollController;
+
+  Future<void> _navigateToBottom(BuildContext context) async {
+    final distance = scrollController.position.pixels;
+    // Calculate time necessary if we're capping speed at 8pixels/ms (== 8000pixels/s)
+    final speedLimit = (distance / 8).ceil();
+    final int durationMs = max(300, speedLimit);
+    scrollController.animateTo(0, duration: Duration(milliseconds: durationMs), curve: Curves.easeIn);
```

Comment:
Can you tell me about your reasoning for picking `Curves.easeIn`?

Reply:
I hate to admit I didn't review all the curve behaviors fully. I did notice that the web behavior is to jump to the bottom and the mobile behavior seems linear. In my previous work with UI I've always found ease-type functions to feel more natural. I must admit my  300ms timer was also a sort of "default practice" for UI based on previous work with animation timings. To evaluate this topic more, I've looked over the material design guidance for motion, see https://m2.material.io/design/motion/speed.html#controlling-speed and https://m3.material.io/styles/motion/easing-and-duration/applying-easing-and-duration and while I got the 300ms perfectly right for their recommended default animation they both recommend a quick-start and slow down for easing, which I believe maps more to the `Curves.ease` rather than `Curves.easeIn`.

Reply:
Cool. Yeah, that fast-start, slow-finish curve sounds good.